src/transformers/vocabulary2id.py

```diff
@@ -0,0 +1,71 @@
+import operator
+
+from pyspark import RDD, Row
+from models.code2vec import Code2Vec
+
+from ast import literal_eval as make_tuple
+from sourced.ml.transformers import Transformer
+
+
+class Vocabulary2Id(Transformer):
+    def __init__(self, keep_freq: bool, output: str, **kwargs):
+        super().__init__(**kwargs)
+        self.keep_freq = keep_freq
+        self.output = output
+
+    def __call__(self, rows: RDD):
+        values, paths = self.build_vocabularies(rows)
+
+        doc2path_contexts = self.build_doc2pc(rows)
+
+        values = values.collect()
+        paths = paths.collect()
+
+        doc2path_contexts = doc2path_contexts.collect()
+
+        Code2Vec().construct(values=values, paths=paths, path_contexts=doc2path_contexts).save(
+            self.output)
+
+    @staticmethod
+    def _unstringify_path_context(row):
+        """
+        Takes a row containing ((pc, doc), freq) and returns a tuple (removes namespace prefix v.)
+        """
+        return make_tuple(row[0][0][2:])
+
+    def _get_path(self, row: Row):
```

Comment:
*_flatten_row

Reply:
_get_path should be renamed to _flatten_row you mean?

Reply:
yeah, `_get_path` is way too generic and impliex you're only getting the path part of the context-path